VoodooI2C/VoodooI2C/VoodooI2CDevice/VoodooI2CDeviceNub.cpp

```diff
@@ -72,20 +71,120 @@ IOReturn VoodooI2CDeviceNub::enableInterrupt(int source) {
     }
 }
 
+IOReturn VoodooI2CDeviceNub::evaluateDSM(const char *uuid, UInt32 index, OSObject **result) {
+    IOReturn ret;
+    uuid_t guid;
+    uuid_parse(uuid, guid);
+
+    // convert to mixed-endian
+    *(reinterpret_cast<uint32_t *>(guid)) = OSSwapInt32(*(reinterpret_cast<uint32_t *>(guid)));
+    *(reinterpret_cast<uint16_t *>(guid) + 2) = OSSwapInt16(*(reinterpret_cast<uint16_t *>(guid) + 2));
+    *(reinterpret_cast<uint16_t *>(guid) + 3) = OSSwapInt16(*(reinterpret_cast<uint16_t *>(guid) + 3));
+
+    OSObject *params[] = {
+        OSData::withBytes(guid, 16),
+        OSNumber::withNumber(I2C_DSM_REVISION, 32),
+        OSNumber::withNumber(index, 32),
+        OSArray::withCapacity(1),
```

Comment:
Finally getting rid of this warning!
Wanted to do it for a long time, but don't have an HID device :P